lxd/db/storage_volumes.go

```diff
@@ -40,21 +40,22 @@ func (c *Cluster) GetStoragePoolVolumesNames(poolID int64) ([]string, error) {
 // GetStoragePoolVolumesWithType return a list of all volumes of the given type.
 func (c *Cluster) GetStoragePoolVolumesWithType(volumeType int) ([]StorageVolumeArgs, error) {
 	var id int64
+	var nodeID sql.NullInt64
```

Comment:
Also returning -1 rather than 0 will help to identify from the default unpopulated value, which may be returned from other functions that return StorageVolumeArgs but won't (yet at least) populate the NodeID field (such as `GetExpiredStorageVolumeSnapshots` and `GetCustomVolumesInProject`.